main/ZgatewayBT.ino

```diff
@@ -819,15 +819,19 @@ void RemoveJsonPropertyIf(JsonObject& obj, char* key, bool condition) {
   }
 }
 
-double value_from_service_data(const char* service_data, int offset, int data_length) {
-  char rev_data[data_length + 1];
+double value_from_service_data(const char* service_data, int offset, int data_length, bool reverse) {
```

Comment:
Why does this function return double type instead of long? strtol returns long. It's problematic for bitwise operation on result (cast to long needed).

Reply:
Thanks for pointing that, I will take a look